src/EA.Iws.Domain.Tests.Unit/Movement/RejectMovementTests.cs

```diff
@@ -0,0 +1,118 @@
+﻿namespace EA.Iws.Domain.Tests.Unit.Movement
+{
+    using Core.Movement;
+    using Domain.Movement;
+    using FakeItEasy;
+    using Prsd.Core;
+    using System;
+    using System.Threading.Tasks;
+    using TestHelpers.DomainFakes;
+    using Xunit;
+
+    public class RejectMovementTests
+    {
+        private static readonly Guid NotificationId = new Guid("675287E3-42D3-4A58-86D4-691ECF620671");
+        private static readonly Guid MovementId = new Guid("DD7A435F-BB74-4EA6-8680-C1811C46500A");
+        private static readonly DateTime Today = new DateTime(2017, 07, 19);
+        private static readonly DateTime PastDate = Today.AddDays(-2);
+        private static readonly DateTime FutureDate = Today.AddDays(2);
+        private readonly IMovementRepository movementRepository;
+        private readonly IMovementRejectionRepository movementRejectionRepository;
+        private readonly IRejectMovement rejectFactory;
+        private readonly String rejectionreason = "Rejection Date validation tests";
+        private Movement movement;
+
+        public RejectMovementTests()
```

Comment:
You need to implement IDisposable and call SystemTime.Unfreeze() in the Dispose() method